### PR TITLE
Updates to help editor

### PR DESCRIPTION
**Pull request by jacklehamster**

### CODE REVIEW

**Comment by jacklehamster** [comment]
Update fixing retina display

**Review by leobenkel — DISMISSED** [review]

**Comment by leobenkel** [comment]
Merge conflict here @jacklehamster

**Comment by jacklehamster** [comment]
Resolved conflicts

File: js/app/game.js [hunk 1]
```diff
@@ -263,6 +263,10 @@ define([
         }
 
         var changeScene = (sceneKey) => {
+            if(!this.scenes[sceneKey]) {
```
Comment (leobenkel) [inline js/app/game.js]: Can you use the isSceneExist method here?

File: js/app/game.js [hunk 2]
```diff
@@ -290,6 +294,14 @@ define([
                 render: true,
             });
         }
+        var dropItem = (itemKey) => {
```
Comment (leobenkel) [inline js/app/game.js]: space above

File: js/app/game.js [hunk 3]
```diff
@@ -290,6 +294,14 @@ define([
...
+            this.isValidItemKey(itemKey, true);
+            this.items[itemKey].owned = false;
+            this.mouse.updateCursor('default');
```
Comment (leobenkel) [inline js/app/game.js]: why do you need this ?

Comment (leobenkel) [reply js/app/game.js]: The rendering is doing that already, no ?

File: js/utility/data-provider.js [hunk 4]
```diff
@@ -1,4 +1,4 @@
-define(['./read-file.js'], function(ReadFile) {
+define(['../utility/read-file.js'], function(ReadFile) {
```
Comment (leobenkel) [inline js/utility/data-provider.js]: why?

File: js/app/location.js [hunk 5]
```diff
@@ -47,8 +48,8 @@ define([
                 }
             },
             draw: (renderer, color) => {
-                var topLeftCorner = renderer.convertCoordonateToBackground(data.description.topLeftCorner);
-                var bottomRightCorner = renderer.convertCoordonateToBackground(data.description.bottomRightCorner);
+                var topLeftCorner = renderer.convertCoordonateToBackground(data.description.topLeftCorner, true);
+                var bottomRightCorner = renderer.convertCoordonateToBackground(data.description.bottomRightCorner, true);
                 var canvasContext = renderer.getContext();
```
Comment (leobenkel) [inline js/app/location.js]: what are all the "true" for ?

File: js/app/game.js (hunk 1 above)
Comment (jacklehamster) [reply js/app/game.js]: I can't find that function

File: js/app/game.js (hunk 3 above)
Comment (jacklehamster) [reply js/app/game.js]: The editor needs a way to remove items.

File: js/app/location.js (hunk 5 above)
Comment (jacklehamster) [reply js/app/location.js]: For displaying: When we use the number for display, they all have to be doubled on retina display, which is different when using the number for mouseover.

File: js/utility/data-provider.js (hunk 4 above)
Comment (jacklehamster) [reply js/utility/data-provider.js]: No need. Removed.

File: js/app/game.js (hunk 2 above)
Comment (jacklehamster) [reply js/app/game.js]: Done

File: js/app/game.js (hunk 1 above)
Comment (leobenkel) [reply js/app/game.js]: https://github.com/The-Brains/ClickAndPointLib/blob/0623a4a363da10dac9f88ad41d083857c81aaea3/js/app/game.js#L181-L194

File: js/app/location.js (hunk 5 above)
Comment (leobenkel) [reply js/app/location.js]: hmm ok, but i didnt see this argument being added to the methods. did I miss it ?

**Review by leobenkel — DISMISSED** [review]

**Comment by jacklehamster** [comment]
Closing. This is a bit too old, I forgot what this is all about.

**Pull request closed without merging**